### PR TITLE
Replace CMake with make (HTCONDOR-360)

### PR DESCRIPTION
**Pull request by brianhlin**
Adds client/CE/central coll make targets, which will be a precursor for separate container images for dev!

### CODE REVIEW

**Comment by brianhlin** [comment]
EL8 OSG test failures are expected -- they're the ones plaguing the nightlies, too

**Comment by edquist** [comment]
:joy:

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+PYTHON			:= /usr/bin/python3
+
+INSTALL_BIN_DIR		:= usr/bin
+INSTALL_LIB_DIR         := usr/lib
+INSTALL_SHARE_DIR	:= usr/share
+INSTALL_STATE_DIR	:= var
+INSTALL_SYSCONF_DIR	:= etc
+INSTALL_WSGI_DIR        := $(INSTALL_STATE_DIR)/www/wsgi-scripts
+INSTALL_PYTHON_DIR	:= $(shell $(PYTHON) -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib())')
```
Comment (matyasselmeci) [inline Makefile:24]: Apparently distutils is going away in Python 3.12. Use (works in both 3 and 2.7):
```suggestion
INSTALL_PYTHON_DIR	:= $(shell $(PYTHON) -c 'from sysconfig import get_paths; print(get_paths()["purelib"])')
```

**Comment by matyasselmeci** [comment]
No objections to the idea, just wondering what the motivation is.

**Comment by brianhlin** [comment]
1. We don't have any C/C++ in the repo anymore
2. Different targets was really annoying to implement in CMake
3. Different targets make it easier/faster to set up the appropriate dev environments

**Comment by edquist** [comment]
> No objections to the idea, just wondering what the motivation is.

**Comment by edquist** [comment]
> No objections to the idea, just wondering what the motivation is.

The motivation for not using straight bash?  ;)

**Review by matyasselmeci — APPROVED** [review]
LGTM

**Pull request merged**